Export/import plots function added

AlexJeBoyy opened the pull request:
added functions to export/import/clear the plots

AlexJeBoyy committed:
Export/import plots function added

BrookJeynes commented on src/components/planner/PlotPlanner.tsx:
```diff
@@ -93,11 +93,23 @@ export function PlotPlanner({
     const [icons, setIcons] = useState<PlannerIcons>(getIconsFromRetrievedPlan);
 
+    const plotPlanLogging = (newPlotPlan: LocalStoragePlotPlan) => {
+        console.log('Plot item placed:', {
```
What benefit does logging this information have to the end user? If this is important, let's put it behind the debug flag and use `console.debug`

AlexJeBoyy replied:
Sorry forgot to delete that 🤦‍♂️

BrookJeynes commented on src/components/planner/PlotPlanner.tsx:
```diff
@@ -93,11 +93,23 @@ export function PlotPlanner({
     const doubleIconYOffset = 0.35;
 
     function handlePlotPlanChange(newPlotPlan: LocalStoragePlotPlan) {
         setPlotPlan(newPlotPlan);
```
`setPlotPlan` is called twice within this function

BrookJeynes commented on src/components/planner/PlotPlannerData.tsx:
```diff
@@ -0,0 +1,92 @@
+            setPlotPlans(getStoredPlotPlans());
+        };
+
+        window.addEventListener("storage", handleStorageChange);
```
Am i correct in the reason we have an event listener on storage here is so we dont have to pass the state variable through to this function?

AlexJeBoyy committed:
#54 making code more readable and function better

Working on making the code more like the rest of the code and everything should work now

AlexJeBoyy commented on src/components/planner/PlotPlanner.tsx:
Still need to delete this (doesnt work properly wwithout it in the last test i did so when thats fixed ill delete that)

AlexJeBoyy committed:
Combined data scripts into one

AlexJeBoyy committed:
Update App.tsx

AlexJeBoyy committed:
Update Sidebar.tsx

AlexJeBoyy committed:
Merge branch 'combine-scripts'

AlexJeBoyy committed:
Couple of missing things

AlexJeBoyy committed:
Update userData.tsx

BrookJeynes commented:
Hey @AlexJeBoyy, is this ready for review?
